src/stream/stream/mod.rs

```diff
@@ -1802,6 +1804,33 @@ extension_trait! {
             CmpFuture::new(self, other)
         }
 
+        #[doc = r#"
+            Counts the number of elements in the stream.
+
+            # Examples
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::prelude::*;
+            use async_std::stream;
+
+            let s1 = stream::from_iter(vec![0]);
+            let s2 = stream::from_iter(vec![1, 2, 3]);
+
+            assert_eq!(s1.count().await, 1);
+            assert_eq!(s2.count().await, 3);
+            #
+            # }) }
+            ```
+        "#]
+        fn count(self) -> impl Future<Output = Ordering> [CountFuture<Self>]
```

Comment:
https://github.com/async-rs/async-std/pull/368/files#diff-14e289ac8d84db1191a59c7117c5a476R29
Here is usize.

Why did you choose Ordering here?

```suggestion
        fn count(self) -> impl Future<Output = usize> [CountFuture<Self>]
```

Reply:
I tried to keep it as generic as possible first, but I guess there is no point in it.